File: 1_algorithmic_toolbox/week2/python/1_fibonacci_number.py
def fibonacci(n):
    if n <= 1:
        return n

    a, b = 1, 0
    for _ in range(n - 1):
        a = a + b
        b = a - b
    return a


if __name__ == "__main__":
    print(fibonacci(int(input())))

File: 1_algorithmic_toolbox/week2/cpp/1_fibonacci_number.cpp
#include <iostream>

int fibonacci(int n)
{
  if (n <= 1) return n;

  int a { 0 }, b { 1 };
  while (--n) {
    b = a + b;
    a = b - a;
  }
  return b;
}

int main()
{
  int n {};
  std::cin >> n;
  std::cout << fibonacci(n) << '\n';
}
